docs/checklist_templates/eip_testing_checklist_template.md

```diff
@@ -0,0 +1,289 @@
+            - [ ] Initcode of a contract creating transaction.
+            - [ ] Initcode of a contract creating opcode (including itself if opcode creates a contract).
+    - [ ] Transaction context
+        - [ ] If opcode changes behavior depending on particular transaction properties, test using multiple values for each property.
+    - [ ] Block context
+        - [ ] If opcode changes behavior depending on particular block properties, test using multiple values for each property.
+    - [ ] EOF Container Context
+        - [ ] If opcode changes behavior depending on particular EOF container properties, test using multiple values for each property.
+- [ ] Return data
+    - [ ] Verify proper return data buffer modification if the opcode is meant to interact with it, otherwise verify that the return data buffer is unnaffected
+        - [ ] At current call context.
+        - [ ] At parent call context.
+- [ ] Gas usage
+    - [ ] Normal operation gas usage
+        - [ ] Verify gas affectation of each parameter value consumed by the opcode.
+    - [ ] Memory expansion
+        - [ ] Verify that the memory expansion correctly follows the gas calculation
+    - [ ] Out-of-gas during opcode execution
+        - [ ] Verify that attempting to execute the opcode when gas available is 1 less than the required gas results in exeptional abort.
+    - [ ] Out-of-gas during memory expansion
+        - [ ] Verify that the expansion of memory can result in out-of-gas exeptional abort.
+- [ ] Terminating opcode
+    - [ ] If an opcode is terminating, meaning it results in the current call context to end execution, test the following scenarios
+        - [ ] Top-level call termination
+        - [ ] Sub-level call termination
+        - [ ] Initcode termination
+- [ ] Out-of-bounds checks
+    - [ ] Verify if the opcode has out-of-bounds conditions in its parameters and verify:
+        - [ ] Max value for each parameter
+        - [ ] Max value + 1 for each parameter
+- [ ] Exeptional Abort
+    - [ ] Verify behavior that is supposed to cause an exeptional abort, other than stack over or underflow, or out-of-gas errors.
+- [ ] Data portion
+    - [ ] If an opcode has data portion that affects its behavior, verify checklist items with multiple interesting values (E.g. if data portion size is 1 byte, use at least 0x00, 0x01, 0x7F, 0xFF).
+- [ ] Contract creation
+    - [ ] Verify contract is created at the expected address given multiple inputs to the opcode parameters.
+    - [ ] Verify that contract is not created in case of
+        - [ ] Out-of-gas when available gas is less than minimum contract creation stipend.
+        - [ ] Contract creation would result in an address collision with an existing contract or eoa-delegated address.
+- [ ] Fork transition
+    - [ ] Verify that the opcode results in exeptional abort if executed before its activation fork.
```

Comment:
Also verify that opcode is availble at fork block, and in some cases also that it yields the correct value (for instance in EIP-1559 the base fee depends on gas used/target gas of previous block, but previous block technically has no "target gas" so a "magic value"/"initial value" is used here)

Reply:
Absolutely, great point! Added.